mipengine/controller/controller.py

```diff
@@ -212,39 +217,39 @@ def validate_algorithm_execution_request(
             available_datasets_per_data_model=available_datasets_per_data_model,
         )
 
-    async def start_node_registry(self):
-        self._controller_logger.info("starting node registry")
-        self._node_registry.keep_updating = True
-        asyncio.create_task(self._node_registry.update())
-        self._controller_logger.info("started node registry")
+    async def start_node_landscape_aggregator(self):
+        self._controller_logger.info("starting node landscape aggregator")
+        self._node_landscape_aggregator.keep_updating = True
+        asyncio.create_task(self._node_landscape_aggregator.update())
+        self._controller_logger.info("started node landscape aggregator")
 
-    async def stop_node_registry(self):
-        self._node_registry.keep_updating = False
+    async def stop_node_landscape_aggregator(self):
+        self._node_landscape_aggregator.keep_updating = False
 
-    def get_all_datasets_per_node(self):
-        datasets = {}
-        for node in self._node_registry.get_all_local_nodes():
-            datasets[node.id] = node.datasets_per_data_model
-        return datasets
+    def get_datasets_location(self):
+        return self._data_model_registry.datasets_location
 
-    def get_all_available_schemas(self):
-        return self._node_registry.get_all_available_data_models()
+    def get_cdes_per_data_model(self):
+        return self._data_model_registry.data_models
+
+    def get_all_available_data_models(self):
+        return list(self._data_model_registry.data_models.keys())
 
     def get_all_available_datasets_per_data_model(self):
-        return self._node_registry.get_all_available_datasets_per_data_model()
+        return self._data_model_registry.get_all_available_datasets_per_data_model()
 
     def get_all_local_nodes(self):
         return self._node_registry.get_all_local_nodes()
 
     def get_global_node(self):
         global_nodes = self._node_registry.get_all_global_nodes()
-        if global_nodes:
-            return global_nodes[0]
+        if NodeRole.GLOBALNODE in global_nodes:
+            return global_nodes[NodeRole.GLOBALNODE]
 
     def _get_nodes_tasks_handlers(
         self, data_model: str, datasets: List[str]
     ) -> NodesTasksHandlersDTO:
-        global_node = self._node_registry.get_all_global_nodes()[0]
+        global_node = self._node_registry.get_all_global_nodes()["globalnode"]
```

Comment:
Maybe we could hide this complex logic inside the `node_registry` by replacing `get_all_global_nodes()` with `get_global_node()` and letting node_registry decide the globalnode to be used. For now it will always be the 1st one but in the future, if we have more than one globalnodes, it's up to the node_registry to return one.

@apmariglis what do you think?

Reply:
yes just define a `get_global_node()`